lib/nx_signal/convolution.ex

```diff
@@ -0,0 +1,269 @@
+defmodule NxSignal.Convolution do
+  @moduledoc """
+  Convolution functions through various methods.
+
+  Follows the `scipy.signal` conventions.
+  """
+
+  import Nx.Defn
+  import NxSignal.Transforms
+
+  @doc """
+  Computes the convolution of two tensors.
```

Comment:
We should add the definition of the convolution used here with Latex as well. And we can also point to the note in the Nx.conv docs.

The docstring is also missing examples with doctests